### DISCUSSION TITLE
Please handle migrations in a stable way

### DISCUSSION THREAD

**Discussion by mback2k** [dfirtrack/dfirtrack#83] [General]
Hello together,

thanks for creating this wonderful project. I have two little wishes which I would even be willing to implement myself for existing migrations (at least wish 2) once I have the time:

1. Please do not change migrations after they have once been "released". Instead create new ones to not break existing installations. Examples to avoid in the future:
- 70a2ac208ebf2b3126d20a86ad89013d1accc1a9
- d4019b1d74f5c3dc29626a3e8c4b99bc2ebc5a09
- 93469ba81ebe6c23b43712d50e1ba40ba62a7c78
- 211c6312902457f870df0b0153a12aa6a1eb60b9
- 437ea685ef53a793e0b6c47b64b48d168fa96909
- 35759c7eda024959ff5a104065bab36e9b27bb80

2. Avoid `RunSQL` if possible and instead work on the dynamically created model objects available during a migration. Example code can be found in the official documentation:
- https://docs.djangoproject.com/en/3.1/topics/migrations/#data-migrations
- https://docs.djangoproject.com/en/3.1/howto/writing-migrations/

Best regards,
Marc

**Comment by stuhli** [comment]
Hi,

thanks for the feedback.

Regarding 1.)
Basically, DFIRTrack was developed as a "throw-away" instance that was no longer needed after the end of a project. This meant that updates were not to be made during operation. Until the completion of version 1.0.0, basic changes could still occur. However, this should no longer occur in the future. 

Regarding 2.)
Feel free to do so. Help is always welcome! However, it must then be ensured that no retroactive changes are made to existing installations. Is that possible with this? What is the advantage of `RunPython` over `RunSQL`? I don't see any preference in the documentation. But I agree that it probably makes the code cleaner. So feel free to do it if you have capacity.

Thank you

**Comment by stuhli** [comment]
As with b57c8e0f181a685ebf4b0108da406322805b8612 all changes were implemented I'm going to lock this discussion.
